zaza/model.py

```diff
@@ -90,20 +90,24 @@ def get_juju_model():
         await model.disconnect()
 
 
-def get_unit_from_name(unit_name, model):
+def get_unit_from_name(unit_name, model_name=None):
     """Return the units that corresponds to the name in the given model.
 
     :param unit_name: Name of unit to match
     :type unit_name: str
-    :param model: Model to perform lookup in
-    :type model: juju.model.Model
+    :param model_name: Name of the model to perform lookup in
+    :type model_name: string or model.Model()
     :returns: Unit matching given name
     :rtype: juju.unit.Unit or None
+    :raises: UnitNotFound
     """
     app = unit_name.split('/')[0]
     unit = None
     try:
-        units = model.applications[app].units
+        if model_name and isinstance(model_name, type(Model)):
```

Comment:
The overloading of `model_name` to be both a model name (string) and model object feels odd.

Reply:
@fnordahl 
Would you be more comfortable with something like:
```python
def get_unit_from_name(unit_name, model=None, model_name=None):
    """Return the units that corresponds to the name in the given model.

    :param unit_name: Name of unit to match
    :type unit_name: str
    :param model_name: Name of the model to perform lookup in
    :type model_name: string or model.Model()
    :param model: model to perform lookup in
    :type model:  model.Model()
    :returns: Unit matching given name
    :rtype: juju.unit.Unit or None
    :raises: UnitNotFound
    """
    app = unit_name.split('/')[0]
    unit = None
    try:
        if model is not None:
            units = model_name.applications[app].units
        elif model_name is not None:
            units = get_units(application_name=app, model_name=model_name)
    except KeyError:
        msg = ('Application: {} does not exist in current model'.
               format(app))
        logging.error(msg)
        raise UnitNotFound(unit_name)
    for u in units:
        if u.entity_id == unit_name:
            unit = u
            break
    else:
        raise UnitNotFound(unit_name)
    return unit
```

Reply:
:+1: